### PR TITLE
added unit test for analyzer checksum change

### PR DESCRIPTION
**Pull request by heejaechang**
adding unit test for https://github.com/dotnet/roslyn/pull/31438

### CODE REVIEW

**Comment by heejaechang** [comment]
@jinujoseph do I need sign off for unit test only changes?

File: src/Workspaces/CoreTest/Execution/SnapshotSerializationTests.cs [hunk 1]
```diff
@@ -441,6 +441,33 @@ public async Task RoundTrip_Analyzer_Serailization_Desktop_Test()
...
 
+        [Fact]
+        public async Task ShadowCopied_Analyzer_Serailization_Desktop_Test()
+        {
+            using (var tempRoot = new TempRoot())
+            {
+                var hostServices = MefHostServices.Create(
+                MefHostServices.DefaultAssemblies.Add(typeof(Host.TemporaryStorageServiceFactory.TemporaryStorageService).Assembly));
+
+                var workspace = new AdhocWorkspace(hostServices);
```
Comment (sharwell) [inline src/Workspaces/CoreTest/Execution/SnapshotSerializationTests.cs]: ❕ need to dispose of workspace before end of test

**Comment by jinujoseph** [comment]
@heejaechang , don't need approval for test only changes

File: src/Workspaces/CoreTest/Execution/SnapshotSerializationTests.cs (hunk 1 above)
Comment (heejaechang) [reply src/Workspaces/CoreTest/Execution/SnapshotSerializationTests.cs]: AdhocWorkspace doesn't do anything on Dispose.

**Pull request merged**